### PR TITLE
#165305737 HighLight Articles

### PR DESCRIPTION
**Pull request by Ogutu-Brian**
#### What does this PR do?
* Adds highlights for articles
#### Description of Task to be completed?
* Ensure an authorized user can highlight an article and remove highlights
#### How should this be manually tested?
* After cloning the repo:
```
$ git checkout ft-highlight-165305737
$ npm run start:dev 
```
#### Screenshots
image
image

image

#### What are the relevant pivotal tracker stories?
[#165305737](https://www.pivotaltracker.com/story/show/165305737)

### CODE REVIEW

File: src/redux/actions/highlight.action.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+};
+
+const updateHighlightAction = (slug, data, id) => async (dispatch) => {
+  try{
+    const response = await updateHighlightApi (data,slug,id);
+    dispatch(updateHighlightSuccess({
+      data:response.data,
+      id:id
+    }));
+  }catch(error){
```
Comment (IssaIan) [inline src/redux/actions/highlight.action.js]: Catch network error here for all actions. When there is no network, user should receive a message. A toast would suffice.

**Review by IssaIan — CHANGES_REQUESTED** [review]

File: src/redux/actions/highlight.action.js (hunk 1 above)
Comment (Ogutu-Brian) [reply src/redux/actions/highlight.action.js]: Hello @IssaIan ,
I have mitigated that by ensuring that if you are not logged in you can't do an highlight nor an update. That is the error that would be thrown by the backend

**Review by Amoswachira — APPROVED** [review]

File: src/components/highlight/CommentComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Modal } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import '../../styles/style.css';
+
+const HighlightComment = ({ showModal, handleModalClose, section, article,
+  comment, getComment, handleSubmit, baseOffset, extentOffset,
+  containsHighlight, updateComment, currentUser }) => {
+  const { author } = article;
+  const commentedSection = containsHighlight(article, section, baseOffset, extentOffset);
```
Comment (a-braham) [inline src/components/highlight/CommentComponent.jsx]: Would you put the properties in a new line

File: src/redux/actions/highlight.action.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
+import ACTION_CONSTANTS from '../constants/constants';
+import APIS from '../../services/api';
+
+const { getHighlightsApi, createHighlightApi, removeHighlightApi, updateHighlightApi } = APIS;
```
Comment (a-braham) [inline src/redux/actions/highlight.action.js]: Here too

**Review by matthenge — APPROVED** [review]

**Review by a-braham — APPROVED** [review]

File: src/components/highlight/CommentComponent.jsx (hunk 2 above)
Comment (Ogutu-Brian) [reply src/components/highlight/CommentComponent.jsx]: I've implemented that, thanks

File: src/redux/actions/highlight.action.js (hunk 3 above)
Comment (Ogutu-Brian) [reply src/redux/actions/highlight.action.js]: I've implemented that, thanks

**Review by Philipotieno — APPROVED** [review]
Good Job on this task it was a heavy one

**Review by IssaIan — APPROVED** [review]

**Review by Bakley — APPROVED** [review]
Awesome job Brian.

**Pull request merged**